awscli/customizations/cloudformation/artifact_exporter.py

```diff
@@ -659,9 +659,23 @@ def export(self):
 
         self.template_dict = self.export_global_artifacts(self.template_dict)
 
-        for resource_id, resource in self.template_dict["Resources"].items():
+        self.export_resources(self.template_dict["Resources"])
+
+        return self.template_dict
+
+    def export_resources(self, resource_dict):
+        for resource_id, resource in resource_dict.items():
+
+            if resource_id.startswith("Fn::ForEach"):
+                if not isinstance(resource, list) or len(resource) < 3:
+                    raise ValueError("Fn::ForEach with name {0} has invalid format".format(resource_id))
+                if isinstance(resource[2], dict):
+                    self.export_resources(resource[2])
+                continue
 
             resource_type = resource.get("Type", None)
+            if resource_type is None:
```

Comment:
I had added this (removed it from the latest commit) to prevent the need to go through the items in `self.resources_to_export` if the `resource_type` resolves to None (see loop below)

```python
            resource_type = resource.get("Type", None)
            # ...
            for exporter_class in self.resources_to_export:
                if exporter_class.RESOURCE_TYPE != resource_type:
                    continue
                # ...
```

However, looking at the `Resource` class, if `RESOURCE_TYPE` is not defined for a subclass of a Resource, it would also resolve to `None`, thought I don't think we'd every have such a subclass, but it is possible, so opting to not have this change as part of this PR